docs/blocks/index.md

```diff
@@ -1,51 +1,182 @@
+### Placing a Block
 
-[sounds]: ../gameeffects/sounds.md
-[creativetabs]: ../items/index.md#creative-tabs
-[registering]: ../concepts/registries.md#methods-for-registering
+Block placement logic is called from `BlockItem#useOn` (or some subclass's implementation thereof, such as in `PlaceOnWaterBlockItem`, which is used for lily pads). For more information on how the game gets there, see the [Interaction Pipeline][interactionpipeline]. In practice, this means that as soon as a `BlockItem` is right-clicked (for example a cobblestone item), this behavior is called.
+
+- Several prerequisites are checked, for example that you are not in spectator mode, that all required feature flags for the block are enabled or that the block in question is not outside the world border. If at least one of these checks fails, the pipeline ends.
+- `Block#canBeReplaced` is called for the block currently at the position where the block is tried to place. If it returns `false`, the pipeline ends.
+- `Block#getStateForPlacement` is called.
+- `Block#canSurvive` is called with the blockstate obtained in the previous step. If it returns `false`, the pipeline ends.
+- The blockstate is set into the level via a `Level#setBlock` call.
+  - In that `Level#setBlock` call, `Block#onPlace` is called.
+- `Block#setPlacedBy` is called.
+
+### Breaking a Block
+
+Breaking a block is a bit more complex, as it requires time. The process can be roughly divided into four stages: "initiating", "mining", "actually breaking" and "finishing".
+
+- When the left mouse button is clicked, the "initiating" stage is entered. 
+- Now, the left mouse button needs to be held down, entering the "mining" stage. **This stage's methods are called every tick.**
+- If the "continuing" stage is not interrupted (by releasing the left mouse button) and the block is broken, the "actually breaking" stage is entered.
+- Under all circumstances, no matter if the block was actually broken or not, the "finishing" stage is entered.
+
+Or for those who prefer pseudocode:
+
+```java
+leftClick();
+initiatingStage();
+while (leftClickIsBeingHeld()) {
+    miningStage();
+    if (blockIsBroken()) {
+        actuallyBreakingStage();
+        break;
+    }
+}
+finishingStage();
+```
+
+The following subsections further break down these stages into actual method calls.
+
+#### The "Initiating" Stage
+
+- `InputEvent.InteractionKeyMappingTriggered` is fired with the left mouse button and the main hand. If the event is canceled, the pipeline ends.
+- Several prerequisites are checked, for example that you are not in spectator mode, that all required feature flags for the `ItemStack` in your main hand are enabled or that the block in question is not outside the world border. If at least one of these checks fails, the pipeline ends.
+- `PlayerInteractEvent.LeftClickBlock` is fired. If the event is canceled, the pipeline ends.
```

Comment:
Correct me if I'm wrong, but this has some differing behavior depending when you intercept this even on the client or server. If on the client, it will stop the packet from being sent to the server altogether which 'continues' the stage to some degree.

This comment applies to all the `PlayerInteractEvent`s.